elements/lisk-p2p/src/peer/inbound.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import {
+	Peer,
+	PeerConfig,
+	REMOTE_EVENT_MESSAGE,
+	REMOTE_EVENT_RPC_REQUEST,
+	SCServerSocketUpdated,
+} from './base';
+
+import { P2PDiscoveredPeerInfo } from '../p2p_types';
+
+import { SCServerSocket } from 'socketcluster-server';
+
+export const EVENT_CLOSE_INBOUND = 'closeInbound';
+export const EVENT_INBOUND_SOCKET_ERROR = 'inboundSocketError';
+
+export class InboundPeer extends Peer {
+	protected _socket: SCServerSocketUpdated | undefined;
+	protected readonly _handleInboundSocketError: (error: Error) => void;
+	protected readonly _handleInboundSocketClose: (
+		code: number,
+		reason: string,
+	) => void;
+
+	public constructor(
+		peerInfo: P2PDiscoveredPeerInfo,
+		peerSocket: SCServerSocket,
+		peerConfig?: PeerConfig,
+	) {
+		super(peerInfo, peerConfig);
+		this._handleInboundSocketError = (error: Error) => {
+			this.emit(EVENT_INBOUND_SOCKET_ERROR, error);
+		};
+		this._handleInboundSocketClose = (code, reason) => {
+			this.emit(EVENT_CLOSE_INBOUND, {
+				peerInfo,
+				code,
+				reason,
+			});
+		};
+		this._socket = peerSocket;
+		this._bindHandlersToInboundSocket(this._socket);
+	}
+
+	public set socket(scServerSocket: SCServerSocket) {
+		if (this._socket) {
+			this._unbindHandlersFromInboundSocket(this._socket);
```

Comment:
For the inbound peer, there will always be an existing socket so we don't need to do this check.